WordPress/src/test/java/org/wordpress/android/viewmodel/domainregister/DomainSuggestionsViewModelTest.kt

```diff
@@ -0,0 +1,123 @@
+import com.nhaarman.mockito_kotlin.any
+import com.nhaarman.mockito_kotlin.argumentCaptor
+import com.nhaarman.mockito_kotlin.atLeast
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.verify
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.action.SiteAction
+import org.wordpress.android.fluxc.annotations.action.Action
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.network.rest.wpcom.site.DomainSuggestionResponse
+import org.wordpress.android.fluxc.store.SiteStore.OnSuggestedDomains
+import org.wordpress.android.fluxc.store.SiteStore.SuggestDomainsPayload
+
+@RunWith(MockitoJUnitRunner::class)
+class DomainSuggestionsViewModelTest {
+    @Rule
+    @JvmField
+    val rule = InstantTaskExecutorRule()
+    @Mock
+    private lateinit var dispatcher: Dispatcher
+    private var site: SiteModel = SiteModel()
+    private val actionCaptor = argumentCaptor<Action<Any>>()
+
+    private lateinit var viewModel: DomainSuggestionsViewModel
+
+    init {
+        site.name = "Wordpress"
+    }
+
+    @Before
+    fun setUp() {
+        viewModel = DomainSuggestionsViewModel(dispatcher)
+        viewModel.site = site
+    }
+
+    @Test
+    fun onStartFetchingSuggestions() {
+        Assert.assertNull(viewModel.selectedSuggestion.value)
+        Assert.assertNull(viewModel.selectedPosition.value)
+
+        viewModel.start(site)
+        assertFetchSuggestions(site.name)
+    }
+
+    @Test
+    fun onSuggestionsSearchQueryChanged() {
+        val query = "sample site name"
+        viewModel.updateSearchQuery(query)
+        assertFetchSuggestions(query)
+    }
+
+    @Test
+    fun onSuggestionSelectionUpdated() {
+        val suggestions = getMockDomainSuggestions()
+
+        viewModel.onDomainSuggestionsFetched(OnSuggestedDomains("", suggestions))
+
+        val enableChooseDomainObserver = mock<Observer<Boolean>>()
+        viewModel.shouldEnableChooseDomain.observeForever(enableChooseDomainObserver)
+
+        validateSuggestion(suggestions, 1)
+        validateSuggestion(suggestions, 0)
+        // Check Deselection
+        validateSuggestion(suggestions, -1)
+        verify(enableChooseDomainObserver, atLeast(3)).onChanged(any())
+    }
+
+    private fun validateSuggestion(suggestions: List<DomainSuggestionResponse>, selectedPosition: Int) {
+        val selectedSuggestion = suggestions.getOrNull(selectedPosition)
+        viewModel.onDomainSuggestionsSelected(selectedSuggestion, selectedPosition)
+
+        Assert.assertEquals(viewModel.selectedSuggestion.value, selectedSuggestion)
+        Assert.assertEquals(viewModel.selectedPosition.value, selectedPosition)
+    }
+
+    private fun assertFetchSuggestions(query: String) {
```

Comment:
I'm probably nitpicking to much - sorry. What about to renaming this method to `assertFetchSuggestionsInvoked`?